### PR TITLE
Added logic to fetch data and updates to the dashboard

### PR DESCRIPTION
**Pull request by skydonline**
### Changes made:
- Created `routes.py` This will serve as the file for all AJAX requests to update the frontend of the application
- Added a `points` and `streak` column to the user model. This keeps track of the current daily streak and the amount of points the user has.
- Implemented logic for the dashboard to update the streak and points section upon page load and every 10 seconds. The basic format to set up a AJAX request to the backend is the following:
```
function FUNCTION_NAME() {
         $.ajax({
             url: ROUTE_URL,
             method: "GET",
             success: function (data) {
                 // DO SOMETHING
             },
             error: function (error) {
                 console.error("Error fetching user stats:", error);
             },
         });
     }
```
  - `FUNCTION_NAME` is what you decide to name the function. `ROUTE_URL` is the name of the route in the `routes.py` file. It should be the same as the `ROUTE_URL` in `@routes.route(ROUTE_URL, methods=['GET'])`. Inside the `success` block, run the necessary code to render the data.
  - Fixed the issue in the Quiz model that prevented the application from running
  - Fixed the issue in the `user_profile.html` file that prevented the `/profile` route from rendering.
    - It appears that Flask doesn't allow multiple script blocks to be within the same `.html` file.

### Things to note:
- Since there were modifications to the database, all previous versions of the database will be invalid. IF YOU EXPERIENCE ERRORS WHILE REGISTERING/LOGGING IN, PLEASE DELETE `site.db`, AND THEN REGISTER A NEW USER. Otherwise, the application will not work.

### CODE REVIEW

**Comment by zareentk** [comment]
Hi Sky remember to ask for a reviewer next time, but I can review this one for now

File: app/src/models.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ class User(UserMixin, db.Model):
     registration_date = db.Column(db.DateTime, default=datetime.now(timezone(timedelta(hours=-5))))
     favorite_subject = db.Column(db.String(50))
     failed_signin_attempts = db.Column(db.Integer, default=0)
+    points = db.Column(db.Integer, default=0)
```
Comment (zareentk) [inline app/src/models.py]: As previously discussed in the meeting Clara’s model can be used for this, so incorporate it if you have availability

File: app/src/templates/user_profile.html [hunk 2]
```diff
@@ -1,27 +1,55 @@
...
+                console.error("Error fetching user stats:", error);
+            },
+        });
+    }
+
+    $(document).ready(function () {
+        userStats();
+
+        // Update every 10 seconds
+        setInterval(userStats, 10000);
```
Comment (zareentk) [inline app/src/templates/user_profile.html]: This could cause an issue when we eventually migrate to cloud, so try to change this to 30s or even 1 minute

**Review by zareentk — CHANGES_REQUESTED** [review]
There doesn’t seem to be any major issues from a first pass, but just fix the setInterval time and I’ll approve it!

**Comment by skydonline** [comment]
I fixed a few issues that prevented the application from working after merging in Clara's and Alex's PRs. There was an issue in the Quiz model ForeignKey reference, `nullable=False` shouldn't be in that reference. Additionally, Flask doesn't allow multiple script blocks to be in the same `.html` file, so I just combined Alex's and my JavaScript code into the same block. I also updated the time to fetch user stats from the database. I will now be working on using Clara's Points model instead of the points field.

**Comment by zareentk** [comment]
Sounds good, the changes fix all the necessary issues so it’s good to go!

**Review by zareentk — APPROVED** [review]

**Pull request merged**